files/scripts/syncd.sh

```diff
@@ -163,11 +163,21 @@ stop() {
         debug "${TYPE} shutdown syncd process ..."
         /usr/bin/docker exec -i syncd$DEV /usr/bin/syncd_request_shutdown --${TYPE}
 
-        # wait until syncd quits gracefully
-        while docker top syncd$DEV | grep -q /usr/bin/syncd; do
+        # wait until syncd quits gracefully or force syncd to exit after 
+        # waiting for 20 seconds
+        start_in_secs=${SECONDS}
+        end_in_secs=${SECONDS}
+        timer_threshold=20
+        while docker top syncd$DEV | grep -q /usr/bin/syncd \
+                && [[ $((end_in_secs - start_in_secs)) -le $timer_threshold ]]; do
             sleep 0.1
+            end_in_secs=${SECONDS}
         done
```

Comment:
Did you try `timeout wait pgrep`?

Reply:
I tried this method. Actually `pgrep` command needs process name or partial process name as its argument. We are unable to use `pgrep \usr\bin\syncd` from host point of view since `\usr\bin\grep` is the execution path in the syncd container.